e2e_correctness/query_neo_mem.py

```diff
@@ -282,3 +286,121 @@ def neo4j_get_graph(neo4j_driver: neo4j.BoltDriver) -> Graph:
     logger.debug("Building the graph from Neo4j JSON data")
 
     return create_graph_neo4j_json(json_data)
+
+
+# additions for path testing
+def sort_dict(dict):
+    keys = list(dict.keys())
+    keys.sort()
+    sorted_dict = {i: dict[i] for i in keys}
+    if "id" in sorted_dict:
+        sorted_dict.pop("id")
+    return sorted_dict
+
+
+def execute_query_neo4j(driver, query):
+    with driver.session() as session:
+        query = neo4j.Query(query)
+        results = session.run(query).value()
+    return results
+
+
+def path_to_string_neo4j(path):
+    path_string = "PATH: "
+
+    n = len(path.nodes)
+
+    for i in range(0, n):
+        node = path.nodes[i]
+        node_labels = list(node.labels)
+        node_labels.sort()
+        node_props = str(sort_dict(node._properties))
+        path_string += (
+            "(id:"
+            + (
+                str(node.get("id"))
+                + " labels: "
+                + str(node_labels)
+                + " "
+                + str(node_props)
+            )
+            + ")-"
+        )
+
+        if i == n - 1:
+            path_string = path_string[:-1]
+            continue
+
+        relationship = path.relationships[i]
+        rel_props = str(sort_dict(relationship._properties))
+        path_string += (
+            "["
+            + (
+                "id:"
+                + str(relationship.get("id"))
+                + " type: "
+                + relationship.type
+                + " "
+                + rel_props
+            )
+            + "]-"
+        )
+
+    return path_string
```

Comment:
Python += -> very bad
I think it would be much better to create path_string_list -> append to it and return ''.join(path_string_list) at the end
Furthermore:
string = string + string + string + string generally should be replaced with an fstring or .format unless its only 2 strings, so (str + str).